src/codecs/person/index.ts

```diff
@@ -2,25 +2,47 @@
  * @module person
  */
 
-import { Person } from '@stencila/schema'
+import * as stencila from '@stencila/schema'
 // @ts-ignore
 import parseAuthor from 'parse-author'
 // @ts-ignore
 import { parseFullName } from 'parse-full-name'
+import * as vfile from '../../util/vfile'
+import { Encode } from '../..'
+import { is, nodeType } from '@stencila/schema/dist/util'
+import log from '../../log'
+
+export const mediaTypes = ['text/x-person']
+
+export const extNames = ['person']
+
+export async function sniff(content: string): Promise<boolean> {
+  return /^\s*[A-Z][a-z.]*(\s+[A-Z][a-z.]*)+(\s+<[\w-_@.]+>)?\s*$/.test(content)
+}
+
+/**
+ * Decode a `VFile` with string content to a `Person`.
+ *
+ * @param file The `VFile` to decode
+ * @returns A promise that resolves to a `Node`
+ */
+export async function decode(file: vfile.VFile): Promise<stencila.Person> {
+  return decodeSync(await vfile.dump(file))
+}
 
 /**
  * Decode string data into a `Person`.
  *
  * @param data Data to decode
  */
-export function decode(data: string): Person {
+export function decodeSync(data: string): stencila.Person {
   const { name, email, url } = parseAuthor(data)
   const { title, first, middle, last, suffix } = parseFullName(name)
-  const person: Person = { type: 'Person' }
+  const person: stencila.Person = { type: 'Person' }
```

Comment:
Should we be using factory functions for things like this now? (May not be available in this branch?)

Reply:
Yes, definitely agree we should. And yes available in this branch.